internal/minijinja-utils/src/loading.rs

```diff
@@ -0,0 +1,656 @@
+// # MiniJinja Template Dependency Analysis
+//
+// This module performs **static analysis** of MiniJinja templates to discover template
+// dependencies through `include`, `import`, `from`, and `extends` statements.
+//
+// ## Three-Phase Analysis
+//
+// 1. **Parse**: MiniJinja's `machinery::parse` converts template source into an AST
+// 2. **Traverse**: `LoadCollector` walks the AST to find template load statements
+// 3. **Analyze**: `analyse_expr` extracts template names from expressions
+//
+// ## Static Analysis Results
+//
+// - **Static** : `"template.html"` → knows exact template name
+// - **Dynamic** : `template_var` → cannot resolve (variable, function call, etc.)
+// - **Partial** : `["known.html", var]` → knows some values, not all
+//
+// The analysis is conservative: when uncertain, expressions are marked dynamic.
+// Conditional expressions like `"a.html" if x else "b.html"` analyze both branches.
+
+use minijinja::machinery::{self, ast, Span, WhitespaceConfig};
+use minijinja::syntax::SyntaxConfig;
+use minijinja::value::Value;
+use minijinja::{Environment, Error};
+use std::collections::{HashSet, VecDeque};
+use std::path::PathBuf;
+
+use crate::error::{AnalysisError, DynamicLoadLocation, LoadKind};
+
+/// Span information extracted from minijinja AST.
+#[derive(Debug, Clone, Copy)]
+struct SpanInfo {
```

Comment:
Do we actually need this type, or can we just pass around the existing minijinja `Span` type?